openssl/src/x509/tests.rs

```diff
@@ -329,3 +330,21 @@ fn signature() {
     assert_eq!(algorithm.object().nid(), nid::SHA256WITHRSAENCRYPTION);
     assert_eq!(algorithm.object().to_string(), "sha256WithRSAEncryption");
 }
+
+#[cfg(any(all(feature = "v102", ossl102), all(feature = "v110", ossl110)))]
+#[test]
+fn test_verify_cert() {
+    let cert = include_bytes!("../../test/cert.pem");
+    let cert = X509::from_pem(cert).unwrap();
+    let ca = include_bytes!("../../test/root-ca.pem");
+    let ca = X509::from_pem(ca).unwrap();
+
+    let mut store_bldr = X509StoreBuilder::new().unwrap();
+    store_bldr.add_cert(ca);
+    let store = store_bldr.build();
+
+    let store_ctx_bldr = X509StoreContext::builder().unwrap();
+    let store_ctx = store_ctx_bldr.build(&store, &cert, &Stack::new().unwrap()).unwrap();
+
+    store_ctx.verify_cert().unwrap();
```

Comment:
Ah yes... and actually, because I move `self` in the `verify()` method I can't check error subsequent to the call... Perhaps a new wrapping type? Like `X509StoreContextPost(_)` which would maintain the reference after the verify, but allow the `error()` check?

Otherwise we end up with a runtime check on needing something reinitialized.